internal/vendors/service.go

```diff
@@ -40,11 +48,52 @@ func (v *VendorService) GetLocations(ctx context.Context) ([]string, error) {
 	return v.vendorDBAccessor.GetAllLocations(ctx)
 }
 
+func (v *VendorService) BlastEmail(ctx context.Context, vendorIDs []string, template emailTemplate) error {
+	vendors, err := v.vendorDBAccessor.BulkGetByIDs(ctx, vendorIDs)
+	if err != nil {
+		return err
+	}
+
+	errCh := make(chan error, len(vendors))
+	defer close(errCh)
+
+	var wg sync.WaitGroup
+
+	for _, vendor := range vendors {
+		wg.Add(1)
+		go func(vendor Vendor) {
+			defer wg.Done()
+
+			// replaces {{name}} keyword to vendor name
+			bodyWithVendorName := strings.Replace(template.Body, "{{name}}", vendor.Name, -1)
+			err := v.smtpProvider.SendEmail(mailer.Email{
+				From:    v.cfg.SMTP.AuthEmail,
+				To:      []string{vendor.Email},
+				Subject: template.Subject,
+				Body:    bodyWithVendorName,
+			})
+			errCh <- err
+		}(vendor)
+	}
+	wg.Wait()
+
+	for i := 0; i < len(vendors); i++ {
+		if err := <-errCh; err != nil {
+			log.Printf("ERROR: fail sending email %v", err)
+		}
+	}
+	return nil
```

Comment:
great question, i was thinking of aggregating the whole error message into one to show which email has not been sent, but I'm not sure what form the error message from the `net/smtp` package look like (since this is a temporary solution for our SES integration for now)
I can't just return err there cause that would cause the function the miss some error messages which is why I didn't return there

wdyt?

Reply:
Hmmmm, if it's temporary for now, it's okay for make sure to mark it as TODO or whatever so you can fix it next sprint because if it's not changed, the client side will have no idea (or notif) if the email blast fails. 

For the solution, isn't it possible to return list of errors or concated string errors of all error messages in the end, so no error messages will be left?

Reply:
Indeed good concern, I was just wondering how the front end would benefit from the "string" error messages
sending an aggregated string is a bit tricky, since the front end would have to parse the error message as well hwem

I think we could use json instead then

Reply:
I think the client can be notified when the email blast fails asynchronously. For example, there could be a history of email blasts with the status of each email being tracked as well (failed or success)

Reply:
Sure, json would work for now 👌